### PR TITLE
Adding connection policies

### PR DESCRIPTION
**Pull request by tunafish88**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [X] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [X] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [X] PowerShell
      - [X] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/sql/resource-manager/readme.md [hunk 1]
```diff
@@ -142,6 +142,7 @@ input-file:
   - Microsoft.Sql/preview/2021-05-01-preview/VirtualNetworkRules.json
   - Microsoft.Sql/preview/2021-05-01-preview/WorkloadClassifiers.json
   - Microsoft.Sql/preview/2021-05-01-preview/WorkloadGroups.json
+  - Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:145]: can we add it as alphabetical order? 

this poershell command will also generate the update list with alphabetical order.
> These can be regenerated by running the following PowerShell script from this readme file's folder: dir .\Microsoft.Sql\preview\2020-11-01-preview\ -File | Resolve-Path -Relative | % { " - $_".Replace("\", "/") }

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/connectionPolicies/{connectionPolicyName}": {
+      "get": {
```
Comment (ericshape) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:20]: can we use https://www.diffchecker.com/diff to post a link about the difference of this connection policy API contract with https://github.com/Azure/azure-rest-api-specs/blob/main/specification/sql/resource-manager/Microsoft.Sql/stable/2014-04-01/connectionPolicies.json ?

Comment (tunafish88) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:20]: https://www.diffchecker.com/oj1G4vVg

File: specification/sql/resource-manager/readme.md (hunk 1 above)
Comment (tunafish88) [reply specification/sql/resource-manager/readme.md:145]: Yes will do.

**Comment by zhenglaizhang** [comment]
@tunafish88 pls fix lintdiff errors https://github.com/Azure/azure-rest-api-specs/pull/15029#issuecomment-870969106

**Comment by ericshape** [comment]
@zhenglaizhang 
For the SQL side, we have not had the system data property and ARM PM team has decided to ignore them since March 2021. 



Rule | Message
-- | --
❌ R4009 - RequiredReadOnlySystemData | The response of operation:'ServerConnectionPolicies_Get' is defined without 'systemData'. Consider adding the systemData to the response.Location: Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json#L20
❌ R4009 - RequiredReadOnlySystemData | The response of operation:'ServerConnectionPolicies_CreateOrUpdate' is defined without 'systemData'. Consider adding the systemData to the response.Location: Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json#L71

**Comment by zhenglaizhang** [comment]
> @zhenglaizhang
> For the SQL side, we have not had the system data property and ARM PM team has decided to ignore them since March 2021.
> 
> Rule	Message
> ❌ R4009 - RequiredReadOnlySystemData	The response of operation:'ServerConnectionPolicies_Get' is defined without 'systemData'. Consider adding the systemData to the response.Location: Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json#L20
> ❌ R4009 - RequiredReadOnlySystemData	The response of operation:'ServerConnectionPolicies_CreateOrUpdate' is defined without 'systemData'. Consider adding the systemData to the response.Location: Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json#L71
> BTW, this PR is required for the ARM deployment of the corresponding API controllers. But we don't need to merge it now. We can wait for the ARM deployment to complete on all prod.

Sounds good, wait for your ack to merge the PR

**Comment by tunafish88** [comment]
Just leaving a coment that this PR is waiting because SQL deployment is taking some time. Deployment is scheduled to start tonight 9/13 PST , later evening (post 7 pm)

**Comment by ericshape** [comment]
close due to we have another all-in-one PR to cover it: #16184

**Pull request closed without merging**